astropy/wcs/utils.py

```diff
@@ -891,9 +891,13 @@ def _linear_wcs_fit(params, lon, lat, x, y, w_obj):
     w_obj.wcs.crpix = crpix
     lon2, lat2 = w_obj.wcs_pix2world(x, y, 0)
 
-    resids = np.concatenate((lon-lon2, lat-lat2))
-    resids[resids > 180] = 360 - resids[resids > 180]
-    resids[resids < -180] = 360	+ resids[resids < -180]
+    lat_resids = lat-lat2
+    lon_resids = lon-lon2
+    # In case the longitude has wrapped around
+    lon_resids[lon_resids > 180] = 360 - lon_resids[lon_resids > 180]
+    lon_resids[lon_resids < -180] = 360 + lon_resids[lon_resids < -180]
```

Comment:
```suggestion
    lon_resids = np.mod(lon_resids - 180.0, 360.0) - 180.0
```

Reply:
Shouldn't the same be applied to `lat_resids`?

Reply:
This assumes the sign of the residual isn't relevant, and since we're passing it to a least squares, presumably it gets squared and doesn't matter.